applications/GeoMechanicsApplication/custom_utilities/dof_utilities.cpp

```diff
@@ -0,0 +1,120 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include <algorithm>
+
+#include "dof_utilities.h"
+#include "includes/variables.h"
+
+using namespace Kratos;
+
+namespace
+{
+
+const Variable<double>& GetTimeDerivativeOf(const Dof<double>* pDof)
+{
+    // Member function `GetVariable` of class `Dof<double>` returns a reference to a `VariableData`
+    // instance. However, that class does not give us access to the first time derivative.
+    // Therefore, we need to downcast to a `Variable<double>` first. In practice, we expect this
+    // cast to always succeed, but we will check it here, just in case.
+    auto p_variable = dynamic_cast<const Variable<double>*>(&pDof->GetVariable());
```

Comment:
For readability, it might be nice to add parentheses here (at first I was wondering why we need the adres of pDof, but it's the pointer to the variable).
```suggestion
    auto p_variable = dynamic_cast<const Variable<double>*>(&(pDof->GetVariable()));
```

Reply:
Yes, I agree that parentheses make the intention clearer. I have added them.